core/src/com/mygdx/game/models/AttackAction.java

```diff
@@ -2,23 +2,47 @@
 
 public class AttackAction implements UnitAction {
 
+    private String name;
+    private String texture;
+    private int[] damage;
+    private int maxRange;
+
+    public AttackAction(String name, String texture, int[] damage) {
+        this.name = name;
+        this.texture = texture;
+        this.damage = damage;
+        maxRange = damage.length;
+
+    }
+
     @Override
     public String getActionName() {
-        return null;
+        return name;
     }
 
     @Override
     public String getActionTexture() {
-        return null;
+        return texture;
     }
 
     @Override
-    public boolean isValidAction(Position startPosition, Position endPosition, HexMap map) {
+    public boolean isValidAction(Command action) {
+        if (action.hasLineOfSight() && action.getDistance() <= maxRange && action.getDistance() > 0 && damage[action.getDistance()] > 0) {
+            return true;
+        }
         return false;
     }
 
     @Override
-    public boolean applyAction(Position startPosition, Position endPosition, HexMap map) {
-        return false;
+    public IntermediateBoard applyAction(Command action, IntermediateBoard board) {
```

Comment:
Before you apply action you should double check that it is valid, using the isValid fucntion you created and throw an exception.